src/compile/facet.ts

```diff
@@ -426,81 +426,65 @@ function parseAxisGroup(model: FacetModel, channel: Channel) {
 }
 
 
-function getXAxesGroup(model: FacetModel): VgMarkGroup {
-  const hasCol = !!model.facet().column;
+export function getSharedAxisGroup(model: FacetModel, channel: 'x' | 'y'): VgMarkGroup {
+  const isX = channel === 'x' ;
+  const facetChannel = isX ? 'column' : 'row';
+  const hasFacet = !!model.facet()[facetChannel];
```

Comment:
`facet()` return the whole `Facet` object in FacetSpec.  

`facet(channel)` would return `FieldDef`.